coq/coq.el

```diff
@@ -743,6 +743,18 @@ If locked span already has a state number, then do nothing. Also updates
   ;;   (message "Unknown command, hopes this won't desynchronize ProofGeneral")
   ;;   t))))
 
+(defun coq-cmd-prevents-proof-omission (cmd)
+  "Instanciation for `proof-script-cmd-prevents-proof-omission'.
+This predicate decides whether a command inside a proof might
+have effects outside the proof, which would prohibit omitting the
+proof, see `proof-script-omit-proofs'.
+
+Commands starting lower case are deemed as tactics that have
+proof local effect only. Everything else is checked against the
+STATECH field in the coq syntax data base, see coq-db.el."
+  (if (proof-string-match coq-lowercase-command-regexp cmd)
+      nil
+    (not (coq-state-preserving-p cmd))))
```

Comment:
as suggested in issue https://github.com/ProofGeneral/PG/issues/689#issuecomment-1476649322

I guess the lowercase test is unneeded; why not using something like this?

```suggestion
  (or (proof-string-match coq-keywords-state-changing-commands cmd)
    ; (not (proof-string-match coq-keywords-local-state-changing-commands cmd)) ; TODO
  ))
```

Regarding the TODO line, I rely on @hendriktews's suggestion to add another STATECH value and extend this:

https://github.com/ProofGeneral/PG/blob/8416875696cb0c4283e96fe721d343277882ecea/coq/coq-syntax.el#L1158-L1168